sdk/src/extras/disk.rs

```diff
@@ -417,14 +418,29 @@ pub mod types {
 
             read_result?;
 
-            let mut results = Vec::with_capacity(handles.len());
+            let mut errors = Vec::new();
             for handle in handles {
                 let result = handle.await.map_err(DiskImportError::other)?;
-                results.push(result);
+                if let Err(err) = result {
+                    errors.push(err);
+                }
             }
 
-            if results.iter().any(|x| x.is_err()) {
-                return Err(DiskImportError::other("one of the upload threads failed"));
+            match errors.len() {
+                1 => {
+                    return Err(DiskImportError::context(
+                        "upload task failed",
```

Comment:
I don't think "task" is meaningful to end users.

I'm imaging something like this:

```
Error while uploading the disk image: Error Response: status: 503 Service Unavailable;
```

OR

```
Errors while uploading the disk image:
* error reading at offset 4096: EIEIO
* block upload failed: 503
* block upload failed: 501c3
```
```

Reply:
👍